shell/platform/embedder/embedder.cc

```diff
@@ -0,0 +1,220 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#define FLUTTER_EXPORT __attribute__((visibility("default")))
+
+#include "flutter/shell/platform/embedder/embedder.h"
+#include <type_traits>
+#include "flutter/common/threads.h"
+#include "flutter/shell/platform/embedder/platform_view_embedder.h"
+
+#define SAFE_ACCESS(pointer, member, default_value)                      \
```

Comment:
My best attempt at using a templated inline function was foiled by the fact that I was using pointer-to-members and `offsetof` not liking the same :/